autotest/units/001_one_port/075_counters_rib/gen.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+from scapy.all import *
+from scapy.contrib.mpls import MPLS
+
+
+def write_pcap(filename, *packetsList):
+	if len(packetsList) == 0:
+		PcapWriter(filename)._write_header(Ether())
+		return
+
+	PcapWriter(filename)
+
+	for packets in packetsList:
+		if type(packets) == list:
+			for packet in packets:
+				packet.time = 0
+				wrpcap(filename, [p for p in packet], append=True)
+		else:
+			packets.time = 0
+			wrpcap(filename, [p for p in packets], append=True)
```

Comment:
IMHO the following is better
```python
writer = PcapWriter(filename, append=True)
    
    for packets in packets_list:
        if isinstance(packets, list):
            for packet in packets:
                packet.time = 0
                writer.write(packet)
        else:
            packets.time = 0
            writer.write(packets)
            
    writer.close()
```

We will avoid reopening and closing the pcap file for every packet and we can use `isinstance` to check if an object is a list